### DISCUSSION TITLE
quantization in espnet2

### DISCUSSION THREAD

**Discussion by miziworld** [espnet/espnet#4232] [Q&A]
Hello, I would like to add and use the quantization option in espnet2. 
This is provided by default in espnet1, but I think that it can be used easily if only dynamic quantization is used.
For example, if model_best.pth that has been trained in egs2 exists in PATH,
```
model_ori=torch.load(PATH + "model_best.pth")
--->  It's OK
quantized_model = torch.quantization.quantize_dynamic(                                                                                               model_ori, {torch.nn.Linear}, dtype=torch.qint8                                                                                              )  
---> AttributeError: 'collections.OrderedDict' object has no attribute 'eval'  error occured
---> It is also occured in loading by using load_state_dict function
```
 if I add model.eval() before quantization, 
AttributeError: 'collections.OrderedDict' object has no attribute 'eval'  error occured.

How can i solve it...?

**Comment by pyf98** [comment]
Hi, I think `torch.load` returns the saved weights as a dictionary, not a torch module. The quantization function needs a torch module as input. So, I would suggest first creating a model and then load the pre-trained weights. After that, you can perform quantization.

Can you try to add quantization in the inference code? e.g.
https://github.com/espnet/espnet/blob/master/espnet2/bin/asr_inference.py#L85

**Comment by pyf98** [comment]
Hi, I added an option of dynamic quantization to ESPnet2 in this PR: https://github.com/espnet/espnet/pull/4349

**Reply by miziworld** [reply]
Can you check the inference speed with quantized model?
